src/server/handlers/consents/{ID}/generateChallenge.ts

```diff
@@ -0,0 +1,130 @@
+import { Scope } from '../../../../model/scope'
+
+// Asynchronously deals with generating challenge, updating consent db
+//  and making outgoing PUT consent/{ID} call
+export async function generateChallengeAndPutConsentId (
+  request: Request,
+  consent: Consent,
+  id: string
+): Promise<void> {
+  try {
+    // If there is no pre-existing challenge for the consent id
+    // Generate one and update database
+    if (!consent.credentialChallenge) {
+      // Challenge generation
+      const challengeValue = await challenge.generate()
+
+      // Updating credentials with generated challenge
+      const credential: ConsentCredential = {
+        credentialChallenge: challengeValue,
+        credentialStatus: 'PENDING',
+        credentialType: 'FIDO'
+      }
+
+      consent = await updateConsentCredential(consent, credential)
+    } else if (consent.credentialStatus === 'ACTIVE') {
+      // TODO: Error handling here - dealt with in #355
+      Logger.error('ACTIVE credential consent has requested challenge')
+      return
+    }
+
+    // Retrieve Scopes
+    const scopesRetrieved: Scope[] = await scopeDB.retrieveAll(id)
+    const scopes = convertScopesToExternal(scopesRetrieved)
+
+    // Outgoing call to PUT consents/{ID}
+    await putConsentId(consent, request, scopes)
+  } catch (error) {
+    Logger.push(error)
+    // eslint-disable-next-line max-len
+    Logger.error(`Error: Outgoing call with challenge credential NOT made to  PUT consent/${id}`)
+    // TODO: Decide on error handling HERE -  dealt with in future ticket #355
+  }
+}
+
+/** The HTTP request `POST /consents/{ID}/generateChallenge` is used to create a
+ * credential for the given Consent object. The `{ID}` in the URI should
+ * contain the `{ID}` that was used in the `POST /consents`.
+ * Called by a `PISP`to request a challenge from the `auth-service`, which
+ * will be returned to the PISP via `PUT /consents/{ID}`
+ */
+export async function post (
+  request: Request, h: ResponseToolkit): Promise<ResponseObject> {
+  const id = request.params.id
+
+  // Fetch consent using ID
+  let consent: Consent
+  try {
+    consent = await consentDB.retrieve(id)
+  } catch (error) {
+    // TODO: Error Handling dealt with in future ticket #355
+    Logger.push(error)
+    Logger.error('Error in retrieving consent')
+
+    // If consent cannot be retrieved using given ID
+    // Return 400 code
+    return h.response().code(400)
+  }
+
+  // If consent is invalid, return 400 code
+  if (!isConsentRequestInitiatedByValidSource(request, consent)) {
+    return h.response().code(400)
+  }
+
+  // Asynchronously deals with generating challenge, updating consent db
+  //  and making outgoing PUT consent/{ID} call
+  generateChallengeAndPutConsentId(request, consent, id)
+  // intentionally not await-ing we want to run it in background
+
+  // Return Success code informing source: request received
+  return h.response().code(202)
```

Comment:
```suggestion
  return h.response().code(Enum.Http.ReturnCodes.ACCEPTED.CODE)
```

and add at the begin of file:
`import { Enum } from '@mojaloop/central-services-shared'`
Use always status error codes from this Enum